specification/compute/resource-manager/Microsoft.Compute/stable/2022-01-03/GalleryRP/communityGallery.json

```diff
@@ -0,0 +1,505 @@
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/communityGalleries/{publicGalleryName}/images/{galleryImageName}/versions": {
+      "get": {
+        "tags": [
+          "CommunityGalleryImageVersions"
+        ],
+        "operationId": "CommunityGalleryImageVersions_List",
+        "description": "List community gallery image versions inside an image.",
+        "parameters": [
+          {
+            "$ref": "../common.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./galleryRPCommon.json#/parameters/LocationNameParameter"
+          },
+          {
+            "$ref": "../common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/PublicGalleryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/GalleryImageNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CommunityGalleryImageVersionList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../common.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List community gallery image versions.": {
+            "$ref": "./examples/communityGalleryExamples/CommunityGalleryImageVersion_List.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "PirCommunityGalleryResource": {
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name"
+        },
+        "location": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource location"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "identifier": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CommunityGalleryIdentifier"
+        }
+      },
+      "description": "Base information about the community gallery resource in pir."
+    },
+    "CommunityGalleryIdentifier": {
+      "properties": {
+        "uniqueId": {
+          "type": "string",
+          "description": "The unique id of this community gallery."
+        }
+      },
+      "description": "The identifier information of community gallery."
+    },
+    "CommunityGallery": {
+      "properties": {},
+      "allOf": [
+        {
+          "$ref": "#/definitions/PirCommunityGalleryResource"
+        }
```

Comment:
I am not sure if we could change this, but since this is returned by a Get API like a resource - could we make this model also like a resource by adding a `id`, `name` and `type` in this model as well as to make the service could also return these values?
Similar requests to CommunityGalleryImage/Version and SharedGallery, Image and Version.
I once wrote an email about this particular subject. The new DotNet track 2 SDK has a resource hierarchy structure - making this a resource will help the generator to produce a SDK with better interface.
This is only a suggestion

Reply:
The PirCommunityGalleryResource model has Name and Type under Properties. Is that not correct? Do you want us to add Id to that? It does have Identifier.
I'm not sure I recall the context, but we can consider doing that on the API side. I'll reach out to you on teams, but we won't do that in this release.